src/main/java/org/elasticsearch/cluster/metadata/AliasFieldsFiltering.java

```diff
@@ -0,0 +1,137 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentBuilderString;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ */
+public class AliasFieldsFiltering implements Streamable, ToXContent {
+
+    private String[] includes;
+
+    public AliasFieldsFiltering(StreamInput in) throws IOException {
+        readFrom(in);
+    }
+
+    public AliasFieldsFiltering(String[] includes) {
+        this.includes = includes;
+    }
+
+    public String[] getIncludes() {
+        return includes;
+    }
+
+    public void setIncludes(String[] includes) {
+        this.includes = includes;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        AliasFieldsFiltering that = (AliasFieldsFiltering) o;
+        return Arrays.equals(includes, that.includes);
+    }
+
+    @Override
+    public int hashCode() {
+        return Arrays.hashCode(includes);
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        includes = in.readStringArray();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeStringArrayNullable(includes);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(Fields.FIELDS);
+        if (includes != null) {
+            builder.startArray(Fields.INCLUDES);
+            for (String field : includes) {
+                builder.value(field);
+            }
+            builder.endArray();
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public static AliasFieldsFiltering fromXContext(XContentParser parser) throws IOException {
+        if (parser.currentToken() != XContentParser.Token.START_OBJECT) {
+            throw new ElasticsearchParseException("Illegal start");
```

Comment:
obscure error message... let's start by "could not read alias fields filtering from xcontent. expected an object but found [" + parser.currentToken() + "] instead"